doc/specs/VideoGIFSpec.md

```diff
@@ -0,0 +1,303 @@
+| Full Screen Capture                        |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |
+| Explicit Recording Dimensions Capture      |![N](./images/GIFSpec/N.png "No") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![N](./images/GIFSpec/N.png "No") |![N](./images/GIFSpec/N.png "No") |![Y](./images/GIFSpec/Y.png "Yes") |
+| Trim Recording                             |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![N](./images/GIFSpec/N.png "No") |![Y](./images/GIFSpec/Y.png "Yes") |
+| Add Text Overlay to Recording              |![N](./images/GIFSpec/N.png "No") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![N](./images/GIFSpec/N.png "No") |![Y](./images/GIFSpec/Y.png "Yes") |
+| System Audio                               |![Y](./images/GIFSpec/Y.png "Yes") |![N](./images/GIFSpec/N.png "No") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![N](./images/GIFSpec/N.png "No") |![Y](./images/GIFSpec/Y.png "Yes") |
+| Optional Cursor Capture                    |![N](./images/GIFSpec/N.png "No") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![N](./images/GIFSpec/N.png "No") |![Y](./images/GIFSpec/Y.png "Yes") |
+| Keystroke Capture                          |![N](./images/GIFSpec/N.png "No") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![N](./images/GIFSpec/N.png "No") |![N](./images/GIFSpec/N.png "No") |![Y](./images/GIFSpec/Y.png "Yes") |
+| Save to MP4                                |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![N](./images/GIFSpec/N.png "No") |![Y](./images/GIFSpec/Y.png "Yes") |
+| Save to GIF                                |![N](./images/GIFSpec/N.png "No") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![N](./images/GIFSpec/N.png "No") |![Y](./images/GIFSpec/Y.png "Yes") |
+| Copy Captured Media to Clipboard           |![N](./images/GIFSpec/N.png "No") |![Y](./images/GIFSpec/Y.png "Yes") |![N](./images/GIFSpec/N.png "No") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |
+| Microphone Audio                           |![Y](./images/GIFSpec/Y.png "Yes") |![N](./images/GIFSpec/N.png "No") |![Y](./images/GIFSpec/Y.png "Yes") |![Y](./images/GIFSpec/Y.png "Yes") |![N](./images/GIFSpec/N.png "No") |![N](./images/GIFSpec/N.png "No") |
```

Comment:
I'm curious why you don't want to support microphone audio recording too.

This is incredibly handy for explaining something when you're trying to demonstrate a problem that you're making a capture for. 

I think it's a small enough thing that you really should consider putting it in scope.

Reply:
Why we think this is a stretch goal is this can have a lot of configuration.

Do you want microphone on this capture? all captures by default?  Which microphone?  Who is handling the levels, the OS or this tool?

Reply:
Audio input is very important in such a tool. It's a Power Toy, right?

I really keep an eye on a tool where the screen is capture and a Webcam / other cam can be captured as well and you can choose that the shape of this second screen can have rounded corners and some shadow effects. That's the perfect tool for tutorials on Windows 😊👍

Reply:
@crutkas 

> Do you want microphone on this capture? all captures by default?
Since there is going to be a selection of whether you want to capture a click-and-drag rectangle, a window or full screen, in the same UI a toggle for mic on/off could be included. For keyboard shortcuts, an alternative key could be used to initiate a recording with mic. (ctrl -> alt)
> Which microphone?
Whichever Audio input device is set as Default Communication Device can be set by default. Additionally, a drop-down in settings to set a different device.
> Who is handling the levels, the OS or this tool?
I would say the OS.

Reply:
I'd be happy with 'default communications mic'. 

I have a mic that has it's own gain, so I'd be ok with levels set by the OS. 

I think the only significant option is 'mic on' or 'mic off' and default to off.

Alternatively, this could be handled by a hotkey just turning on mic passthru (since then it'll go thru the desktop audio).